### PR TITLE
avm2: Implement XMLList.copy()

### PR DESCRIPTION
**Pull request by yoganlava**
First PR, open for feedback.

Progresses: #10192

### CODE REVIEW

**Comment by Dinnerbone** [comment]
Welcome and thank you!

Would you be able to make a test for this, or would you like some help with that?

**Comment by yoganlava** [comment]
No problem. Would you like me to add it to the existing xml_basic test or have it exist on it's own?

**Comment by n0samu** [comment]
I'm pretty sure either one would be fine!

**Comment by n0samu** [comment]
That's fine, but could you keep the code inside the run() function of the Test class? The FLA has almost nothing in it, just this code on frame 1:
`Test.run();`
I think you could do the same in your AS file, thus avoiding the need to move all of the code outside the run() function.

File: core/src/avm2/globals/xml_list.rs [hunk 1]
```diff
@@ -232,3 +232,13 @@ pub fn text<'gc>(
...
     Ok(XmlListObject::new(activation, nodes, Some(xml_list.into())).into())
 }
+
+pub fn copy<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let list = this.unwrap().as_xml_list_object().unwrap();
+    let children = (*list.children()).clone();
```
Comment (Bale001) [inline core/src/avm2/globals/xml_list.rs:242]: I don't think this is correct - the [docs](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/XMLList.html#copy()) say `The copy is a duplicate of the entire tree of nodes.`, but this only clones the first layer.

**Comment by yoganlava** [comment]
Closing this pull request as I might still be too new to impl a recursive deep_clone

**Pull request closed without merging**